Update Dependencies - Boost Security and keep repo up to date

dr-bizz opened the pull request:
## Description
In this PR, I upgraded packages that Dependabot has flagged and created a PR for us to merge. In each commit I upgrade a package or improve code.

### Big changes:
1. upgrading `rxjs` to version  7.
2. Updating `eslint` to version 9. _- Doing this meant I had to use the new configuration file. I also added a few settings which we devs are used to working with._

### Things to be aware of
1. Updating @rails/actioncable, we use this when mobile wants to share what they are viewing on the web. _- I have tested this, but it would be good for someone to double test this._
2. Added Husky `pre-commit` commands to ensure prettier and lint test pass.

canac commented on package.json:
```diff
@@ -23,50 +25,44 @@
     "karma-jasmine-html-reporter": "^1.7.0",
     "prettier": "^3.0.0",
     "pretty-quick": "^3.1.0",
-    "stream": "^0.0.3",
```
Are you sure `stream` isn't needed? When I run `yarn build`, I see this error:

```
./node_modules/sax/lib/sax.js:139:13-37 - Warning: Module not found: Error: Can't resolve 'stream' in '/Users/caleb/dev/know-god-web/node_modules/sax/lib'
```

canac reviewed (commented):
I have some questions, but this looks really good! I ran `yarn start` and clicked through a couple of the tools, and everything looked good (besides a `streams` warning that I already mentioned). Is there anything else that I should test?

dr-bizz commented on tsconfig.json:
```diff
@@ -11,6 +11,10 @@
     "typeRoots": ["node_modules/@types"],
-    "lib": ["es2017", "dom"]
+    "lib": ["es2017", "dom"],
+    "paths": {
```
I was getting errors on these before. I can test without it again

dr-bizz replied:
Seems like it's working now. Weird, I must have fixed the error a different way.

dr-bizz commented:
@canac, I have updated the code to match the things you mentioned. I also rebased the commit, but before doing so, I have taken screenshots of the code I changed, so you don't have to review all of it again.

![Screenshot 2024-11-21 at 9 15 31 AM](https://github.com/user-attachments/assets/7e5d5980-c144-415a-9c36-f2d31a656d98)

![Screenshot 2024-11-21 at 9 15 41 AM](https://github.com/user-attachments/assets/6d411b07-bac2-4864-a77d-426e70deefa2)

![Screenshot 2024-11-21 at 9 15 48 AM](https://github.com/user-attachments/assets/6015dd7a-e51a-4e52-8218-746fbbfe94cd)

![Screenshot 2024-11-21 at 9 15 56 AM](https://github.com/user-attachments/assets/fd95a7d6-90b4-45e1-aded-370d9b475105)

canac commented on package.json:
```diff
@@ -8,11 +8,13 @@
     "build:dev": "ng build --configuration development",
     "test": "ng test --watch=false",
-    "lint": "npx eslint 'src/**/*.{js,ts}' --quiet --fix",
+    "lint": "yarn eslint 'src/**/*.{js,ts,html}' --fix",
```
You shouldn't need the `yarn` prefix either.

```suggestion
    "lint": "eslint 'src/**/*.{js,ts,html}' --fix",
```

canac commented on .vscode/settings.json:
```diff
@@ -0,0 +1,13 @@
+  "eslint.validate": ["javascript", "typescript", "html"],
+  "editor.formatOnSave": true,
+  "typescript.tsdk": "node_modules/typescript/lib",
+  "eslint.nodePath": "node_modules",
+  "prettier.prettierPath": "./node_modules/prettier",
```
Are you sure these two paths are needed? I'd think that it would default to looking for the eslint and prettier executables in `node_modules`.

dr-bizz replied:
Yes, we need them, otherwise when I save a file, I get these prettier errors

```
"ERROR" - 4:21:49 PM] this.prettierModule.resolveConfigFile is not a function
TypeError: this.prettierModule.resolveConfigFile is not a function
```

canac reviewed (approved):
I think you need to run `yarn` again and commit the changes to the lockfile. But after that, this all looks pretty good! Thanks for updating everything!
